### PR TITLE
Update bito-cra.ps1 - default branch and acceptable suggestions

### PR DESCRIPTION
**Pull request by rishabhbohra-bito**

### CODE REVIEW

**Comment by BitoAgent** [comment]
## Code Review Agent Run Status

- **Limitations and other issues:** <small>❌</small>&nbsp;Failure - The AI Code Review Agent skipped reviewing this change because it is configured to exclude certain pull requests based on the source/target branch or the pull request status. You can change the settings [here](https://alpha.bito.ai/home/ai-agents/configured-agents/1), or contact the agent instance creator at [nisha.kumari@bito.ai](mailto:nisha.kumari@bito.ai).

**Review by adhirpotdarbito — APPROVED** [review]
Approved.

**Comment by BitoAgent** [comment]
AI Code Review is in progress (usually takes <3 minutes unless it's a very large PR).

**Comment by BitoAgent** [comment]
<div>
<h2>Changelist by <a href="https://bito.ai">Bito</a></h2>
<p>This pull request implements the following key changes.</p>

<div>
<table width="100%">
<tbody>
  <tr>
    <th>Key Change</th>
    <th>Files Impacted</th>
  </tr>

<tr>
<td width="30%">
    <b>Feature Improvement</b> - Enhanced Configuration Options for Code Review Assistant
</td>
<td width="70%">


<div>

-&nbsp;[`bito-cra.ps1`](https://github.com/gitbito/CodeReviewAgent/pull/46/files#diff-b161aef2c7dd34c8e26ada0bb72ed232e46dd9089391163b0531055195230aaf) - Added new parameters for acceptable suggestions and default branch handling
<div>

-&nbsp;[`bito-cra.sh`](https://github.com/gitbito/CodeReviewAgent/pull/46/files#diff-ec48dc0c9bb3ba5472531385ce779efd6309a778d53f9c3a675071e714b2e80b) - Added corresponding parameters in shell script for consistency
</div>
</td>
</tr>

</tbody>
</table>
</div>
</div>

File: cra-scripts/bito-cra.sh [hunk 1]
```diff
@@ -542,7 +546,7 @@ done
...
+  elif [ "$param" != "acceptable_suggestions_enabled" ] && [ "$param" != "dependency_check.snyk_auth_token" ] && [ "$param" != "env" ] && [ "$param" != "cli_path" ] && [ "$param" != "output_path" ] && [ "$param" != "static_analysis_tool" ]  && [ "$param" != "linters_feedback" ] && [ "$param" != "secret_scanner_feedback" ] && [ "$param" != "enable_default_branch" ] && [ "$param" != "git.domain" ] && [ "$param" != "review_scope" ] && [ "$param" != "exclude_branches" ] && [ "$param" != "nexus_url" ] && [ "$param" != "exclude_files" ] && [ "$param" != "exclude_draft_pr" ] && [ "$param" != "cr_event_type" ] && [ "$param" != "posting_to_pr" ] && [ "$param" != "custom_rules.configured_ws_ids" ] && [ "$param" != "custom_rules.aws_access_key_id" ] && [ "$param" != "custom_rules.aws_secret_access_key" ] && [ "$param" != "custom_rules.region_name" ] && [ "$param" != "custom_rules.bucket_name" ] && [ "$param" != "custom_rules.aes_key" ] && [ "$param" != "code_context_config.partial_timeout" ] && [ "$param" != "code_context_config.max_depth" ] && [ "$param" != "code_context_config.kill_timeout_sec" ]; then
```
Comment (BitoAgent) [inline cra-scripts/bito-cra.sh:549]: <div>


<div id="suggestion">
<div id="issue"><b>Consider refactoring long parameter check condition</b></div>
<div id="fix">

The condition check contains a long chain of `!=` comparisons which makes the code hard to maintain and error-prone. Consider using an array or associative array to store these parameter names and using a loop to check if the parameter exists in the array.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -546,7 +546,19 @@
  for param in "${optional_params[@]}"; do
    if [ "$param" == "dependency_check.snyk_auth_token" ] && [ "${props["dependency_check"]}" == "True" ]; then
        ask_for_param "$param" "False"
 -  elif [ "$param" != "acceptable_suggestions_enabled" ] && [ "$param" != "dependency_check.snyk_auth_token" ] && [ "$param" != "env" ] && [ "$param" != "cli_path" ] && [ "$param" != "output_path" ] && [ "$param" != "static_analysis_tool" ]  && [ "$param" != "linters_feedback" ] && [ "$param" != "secret_scanner_feedback" ] && [ "$param" != "enable_default_branch" ] && [ "$param" != "git.domain" ] && [ "$param" != "review_scope" ] && [ "$param" != "exclude_branches" ] && [ "$param" != "nexus_url" ] && [ "$param" != "exclude_files" ] && [ "$param" != "exclude_draft_pr" ] && [ "$param" != "cr_event_type" ] && [ "$param" != "posting_to_pr" ] && [ "$param" != "custom_rules.configured_ws_ids" ] && [ "$param" != "custom_rules.aws_access_key_id" ] && [ "$param" != "custom_rules.aws_secret_access_key" ] && [ "$param" != "custom_rules.region_name" ] && [ "$param" != "custom_rules.bucket_name" ] && [ "$param" != "custom_rules.aes_key" ] && [ "$param" != "code_context_config.partial_timeout" ] && [ "$param" != "code_context_config.max_depth" ] && [ "$param" != "code_context_config.kill_timeout_sec" ]; then
 +  else
 +    skip_params=(
 +      "acceptable_suggestions_enabled" "dependency_check.snyk_auth_token" "env" "cli_path" "output_path"
 +      "static_analysis_tool" "linters_feedback" "secret_scanner_feedback" "enable_default_branch" "git.domain"
 +      "review_scope" "exclude_branches" "nexus_url" "exclude_files" "exclude_draft_pr" "cr_event_type"
 +      "posting_to_pr" "custom_rules.configured_ws_ids" "custom_rules.aws_access_key_id"
 +      "custom_rules.aws_secret_access_key" "custom_rules.region_name" "custom_rules.bucket_name"
 +      "custom_rules.aes_key" "code_context_config.partial_timeout" "code_context_config.max_depth"
 +      "code_context_config.kill_timeout_sec"
 +    )
 +    skip_param=false
 +    for skip in "${skip_params[@]}"; do
 +      [[ "$param" == "$skip" ]] && skip_param=true && break
 +    done
 +    [[ "$skip_param" == false ]] && ask_for_param "$param" "False"

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/gitbito/CodeReviewAgent/pull/46#issuecomment-2536074802>#9f0e9d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: cra-scripts/bito-cra.sh [hunk 2]
```diff
@@ -580,9 +584,15 @@ for param in "${required_params[@]}" "${bee_params[@]}" "${optional_params[@]}";
     elif [ "$param" == "secret_scanner_feedback" ]; then
         props[$param]=$(validate_boolean "${props[$param]}")
         docker_cmd+=" --secret_scanner_feedback=${props[$param]}"
+    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
+        props[$param]=$(validate_boolean "${props[$param]}")
+        docker_cmd+=" --acceptable_suggestions_enabled=${props[$param]}"
```
Comment (BitoAgent) [inline cra-scripts/bito-cra.sh:589]: <div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating boolean parameter handling</b></div>
<div id="fix">

Consider consolidating the boolean parameter handling into a helper function to reduce code duplication. The pattern `validate_boolean` followed by setting docker command is repeated multiple times. A similar issue was also found in cra-scripts/bito-cra.ps1 (line 570-581).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -584,15 +584,7 @@
  function handle_boolean_param() {
    local param=$1
    local value=$2
    props[$param]=$(validate_boolean "$value")
    docker_cmd+=" --$param=${props[$param]}"
  }

      elif [ "$param" == "secret_scanner_feedback" ]; then
 -        props[$param]=$(validate_boolean "${props[$param]}")
 -        docker_cmd+=" --secret_scanner_feedback=${props[$param]}"
 -    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
 -        props[$param]=$(validate_boolean "${props[$param]}")
 -        docker_cmd+=" --acceptable_suggestions_enabled=${props[$param]}"
 +        handle_boolean_param "$param" "${props[$param]}"
 +    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
 +        handle_boolean_param "$param" "${props[$param]}"
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/gitbito/CodeReviewAgent/pull/46#issuecomment-2536074802>#9f0e9d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: cra-scripts/bito-cra.ps1 [hunk 3]
```diff
@@ -534,7 +538,7 @@ foreach ($param in $required_params) {
...
+    } elseif ($param -ne "acceptable_suggestions_enabled" -and $param -ne "dependency_check.snyk_auth_token" -and $param -ne "env" -and $param -ne "cli_path" -and $param -ne "output_path" -and $param -ne "static_analysis_tool" -and $param -ne "linters_feedback" -and $param -ne "secret_scanner_feedback" -and $param -ne "enable_default_branch" -and $param -ne "git.domain" -and $param -ne "review_scope" -and $param -ne "exclude_branches" -and $param -ne "exclude_files" -and $param -ne "exclude_draft_pr" -and $param -ne "cr_event_type" -and $param -ne "posting_to_pr" -and $param -ne "custom_rules.configured_ws_ids"  -and  $param -ne "custom_rules.aws_access_key_id"  -and  $param -ne "custom_rules.aws_secret_access_key"  -and  $param -ne "custom_rules.region_name"  -and  $param -ne "custom_rules.bucket_name"  -and  $param -ne "custom_rules.aes_key"  -and  $param -ne "code_context_config.partial_timeout"  -and  $param -ne "code_context_config.max_depth"  -and  $param -ne "code_context_config.kill_timeout_sec") {
```
Comment (BitoAgent) [inline cra-scripts/bito-cra.ps1:541]: <div>


<div id="suggestion">
<div id="issue"><b>Consider restructuring long condition statement</b></div>
<div id="fix">

Consider breaking down the long condition into multiple lines or extracting it into a helper function for better readability. The current line exceeds recommended line length and makes code maintenance difficult.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -534,11 +534,20 @@
  function Should-Prompt-Parameter($param) {
     $excluded_params = @(
         "acceptable_suggestions_enabled", "dependency_check.snyk_auth_token", "env", "cli_path", 
         "output_path", "static_analysis_tool", "linters_feedback", "secret_scanner_feedback", 
         "enable_default_branch", "git.domain", "review_scope", "exclude_branches", "exclude_files", 
         "exclude_draft_pr", "cr_event_type", "posting_to_pr", "custom_rules.configured_ws_ids", 
         "custom_rules.aws_access_key_id", "custom_rules.aws_secret_access_key", "custom_rules.region_name", 
         "custom_rules.bucket_name", "custom_rules.aes_key", "code_context_config.partial_timeout", 
         "code_context_config.max_depth", "code_context_config.kill_timeout_sec"
     )
     return $excluded_params -notcontains $param
  }

  foreach ($param in $optional_params) {
      if ($param -eq "dependency_check.snyk_auth_token" -and $props["dependency_check"] -eq "True") {
          Ask-For-Param $param $false
 -    } elseif ($param -ne "acceptable_suggestions_enabled" -and $param -ne "dependency_check.snyk_auth_token" -and $param -ne "env" -and $param -ne "cli_path" -and $param -ne "output_path" -and $param -ne "static_analysis_tool" -and $param -ne "linters_feedback" -and $param -ne "secret_scanner_feedback" -and $param -ne "enable_default_branch" -and $param -ne "git.domain" -and $param -ne "review_scope" -and $param -ne "exclude_branches" -and $param -ne "exclude_files" -and $param -ne "exclude_draft_pr" -and $param -ne "cr_event_type" -and $param -ne "posting_to_pr" -and $param -ne "custom_rules.configured_ws_ids"  -and  $param -ne "custom_rules.aws_access_key_id"  -and  $param -ne "custom_rules.aws_secret_access_key"  -and  $param -ne "custom_rules.region_name"  -and  $param -ne "custom_rules.bucket_name"  -and  $param -ne "custom_rules.aes_key"  -and  $param -ne "code_context_config.partial_timeout"  -and  $param -ne "code_context_config.max_depth"  -and  $param -ne "code_context_config.kill_timeout_sec") {
 +    } elseif (Should-Prompt-Parameter $param) {
          Ask-For-Param $param $false
      }

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/gitbito/CodeReviewAgent/pull/46#issuecomment-2536074802>#9f0e9d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

**Pull request merged**